examples/jsm/geometries/SDFGeometryGenerator.js

```diff
@@ -0,0 +1,311 @@
+
+/**
+ * @author santiago / @glitch_life
+ * wrapper of https://www.npmjs.com/package/isosurface by https://github.com/mikolalysenko 
+ * 
+ * Returns BufferGeometry from SDF
+ * 
+ */
+
+import { 
+		BufferAttribute, 
+		BufferGeometry, 
+		FloatType, 
+		Mesh, 
+		OrthographicCamera, 
+		PlaneGeometry, 
+		Scene, 
+		ShaderMaterial, 
+		Vector2, 
+		WebGLRenderTarget 
+} from 'three';
+
+class SDFGeometryGenerator extends BufferGeometry{
+	constructor(  renderer ){
+		super();
+
+		if( !renderer ) return console.warn( 'From MeshCompute : No renderer provided' );
+
+		this.renderer = renderer;
+		this.maxTexSize = renderer.capabilities.maxTextureSize;
+	}
+
+	generate( res = 64, distFunc = `float dist( vec3 p ){ return length(p) - 0.5; }`, bounds = 1 ){
+		let w,h;
+		if( res == 8 ) [w,h] = [ 32, 16];
+		else if( res == 16 ) [w,h] = [ 64, 64 ];
+		else if( res == 32 ) [w,h] = [ 256, 128 ];
+		else if( res == 64 ) [w,h] = [ 512, 512 ];
+		else if( res == 128 ) [w,h] = [ 2048, 1024 ];
+		else if( res == 256 ) [w,h] = [ 4096, 4096 ];
+		else if( res == 512 ) [w,h] = [ 16384, 8096 ];
+		else return console.warn( 'From MeshCompute : Resolution must be in range 8 < res < 512 and must be ^2');
```

Comment:
Regarding console logs:

- When the code can't proceed because of wrong input, throw a new error so devs can react via try/catch.
- If the code can proceed but e.g. you have to clamp input values or resize buffers, a warning is appropriate.

Reply:
So in this case I would do:
```js
else throw new Error ('THREE. SDFGeometryGenerator: Resolution must be in range 8 < res < 512 and must be ^2' );
```

Reply:
✅